lib/coursemology/polyglot/concrete_language/class_methods.rb

```diff
@@ -4,4 +4,26 @@
 # This is deliberately defined so that consumers of this library can inject methods into all
 # concrete languages' classes.
 module Coursemology::Polyglot::ConcreteLanguage::ClassMethods
+  # The default algorithm for translating a language to a corresponding Docker image name.
+  #
+  # This will:
+  #  1. Strip the +Coursemology::Polyglot::Language+ prefix.
+  #  2. Underscore the name.
+  #  3. Replace *n*Point*n* strings with dots (e.g. Python2Point7 to Python2.7).
+  #  4. Replace slashes with dashes.
+  def docker_image
+    # Remove the Coursemology::Polyglot::Language prefix
+    class_name = name
+    class_name = class_name[34..-1] if class_name.start_with?(Coursemology::Polyglot::Language.name)
```

Comment:
it's actually a substring... hmm, sub would look like:
```ruby
class_name.sub!(/^Coursemology::Polyglot::Language::/, '')
```

if that's better I'll change it?